internal/zinc-scripted/src/test/scala/sbt/internal/inc/IncHandler.scala

```diff
@@ -343,6 +343,10 @@ case class ProjectStructure(
     ()
   }
 
+  def classpath(i: IncInstance): Array[File] = {
+    (i.si.allJars.toList ++ (unmanagedJars :+ classesDir) ++ internalClasspath).toArray
```

Comment:
I suspect this line reveals a potential bug. Classes dir should always come before `unmanagedJars`, since `unmanagedJars` could be overriding classes defined in `classesDir`. I suspect `unmanagedJars` has to come at the end of the classpath.

Reply:
This is for emulating subproject dependencies.

Reply:
???. An elaboration would be good.

Reply:
I was able to reproduce this without multiple subprojects, so I'll send this change in another PR.

Reply:
You have not answered why `unmanagedJars` has to be before `classesDir`.